src/util-memcmp.c

```diff
@@ -35,102 +35,134 @@
 
 static int MemcmpTest01 (void)
 {
-    uint8_t a[] = "abcd";
-    uint8_t b[] = "abcd";
+    uint8_t a[] = "abcd            abcd";
+    uint8_t b[] = "abcd            abcd";
+
+    for (size_t x = 1; x < sizeof(a); x++) {
+        FAIL_IF(SCMemcmp(a, b, sizeof(a) - 1) != 0);
+    }
 
-    FAIL_IF(SCMemcmp(a, b, sizeof(a) - 1) != 0);
     PASS;
 }
 
 static int MemcmpTest02 (void)
 {
-    uint8_t a[] = "abcdabcdabcdabcd";
-    uint8_t b[] = "abcdabcdabcdabcd";
+    uint8_t a[] = "abcdabcdabcdabcd    ";
```

Comment:
Thanks for resubmitting.

Looks like you did not take into account https://github.com/OISF/suricata/pull/8764#discussion_r1398105175

> we shouldn't update existing tests for this, but instead add a new set of tests.

Reply:
I'm happy to duplicate the test if you want, but I did try to follow the spirit of making sure the old tests also were covered. 

While the data is extended here, the for loop below covers the original size and all the way up to the new length, so we're not loosing the original coverage.

Reply:
Indeed, I see your point...
It is harder to review...